File: shared/tracking/src/main/java/de/gfred/shared/tracking/di/TrackingServiceModule.java
package de.gfred.shared.tracking.di;

import de.gfred.shared.models.tracking.TrackingService;
import de.gfred.shared.tracking.TrackingServiceImpl;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Singleton
@Module
public class TrackingServiceModule
{
    @Provides
    public TrackingService provideTrackingService(final TrackingServiceImpl trackingServiceImpl)
    {
        return trackingServiceImpl;
    }
}

File: shared/tracking/src/main/kotlin/de/gfred/shared/tracking/di/TrackingServiceModule.kt
package de.gfred.shared.tracking.di

import dagger.Module
import dagger.Provides
import de.gfred.shared.tracking.TrackingService
import de.gfred.shared.tracking.TrackingServiceImpl
import javax.inject.Singleton

@Singleton
@Module
class TrackingServiceModule {
    @Provides
    fun provideTrackingService(trackingServiceImpl: TrackingServiceImpl): TrackingService = trackingServiceImpl
}
